File: Longest Common Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string> &strs) {
        string prefix;
        if (strs.empty())
            return prefix;
        
        int index = 0;
        while (true)
        {
            if (index >= strs[0].size())
                return prefix;
            char ch = strs[0][index];
            for (int i = 1; i < strs.size(); ++i)
            {
                if (index >= strs[i].length() || ch != strs[i][index])
                    return prefix;
            }
            prefix.push_back(ch);
            index++;
        }
        return prefix;
    }
};

File: Longest Common Prefix.py
class Solution:
    # @return a string
    def longestCommonPrefix(self, strs):
        if (len(strs) == 0):
            return ""
        index = 0
        for ch in strs[0]:
            for str in strs[1:]:
                if (len(str) <= index or str[index] != ch):
                    return strs[0][0:index]
            index = index + 1
        return strs[0][0:index]